lib/router.go

```diff
@@ -39,48 +38,38 @@ const (
 // come from, Route and Backend should not contain bson fields.
 // MongoReplicaSet, MongoReplicaSetMember etc. should move out of this module.
 type Router struct {
-	mux               *triemux.Mux
-	lock              sync.RWMutex
-	mongoReadToOptime bson.MongoTimestamp
-	logger            logger.Logger
-	opts              Options
-	ReloadChan        chan bool
+	mux        *triemux.Mux
+	lock       sync.RWMutex
+	logger     logger.Logger
+	opts       Options
+	ReloadChan chan bool
 }
 
 type Options struct {
-	MongoURL             string
-	MongoDBName          string
-	MongoPollInterval    time.Duration
+	DatabaseURL          string
+	DatabaseName         string
+	Listener             *pq.Listener
+	DatabasePollInterval time.Duration
 	BackendConnTimeout   time.Duration
 	BackendHeaderTimeout time.Duration
 	LogFileName          string
 }
 
 type Backend struct {
-	BackendID     string `bson:"backend_id"`
-	BackendURL    string `bson:"backend_url"`
-	SubdomainName string `bson:"subdomain_name"`
-}
-
-type MongoReplicaSet struct {
-	Members []MongoReplicaSetMember `bson:"members"`
-}
-
-type MongoReplicaSetMember struct {
-	Name    string              `bson:"name"`
-	Optime  bson.MongoTimestamp `bson:"optime"`
-	Current bool                `bson:"self"`
+	BackendID     sql.NullString
+	BackendURL    sql.NullString
+	SubdomainName sql.NullString
 }
 
 type Route struct {
-	IncomingPath string `bson:"incoming_path"`
-	RouteType    string `bson:"route_type"`
-	Handler      string `bson:"handler"`
-	BackendID    string `bson:"backend_id"`
-	RedirectTo   string `bson:"redirect_to"`
-	RedirectType string `bson:"redirect_type"`
-	SegmentsMode string `bson:"segments_mode"`
-	Disabled     bool   `bson:"disabled"`
+	IncomingPath sql.NullString
```

Comment:
optional: it'd be neat if we could use plain old strings here and deal with the nullability in `loadRoutes` or wherever, then we could delete this struct's evil twin in `integration_tests/route_helpers.go`. (Would also mean you wouldn't need to deal with nulls in all the places that use struct Route.)